Lib/warnings.py

```diff
@@ -176,6 +176,7 @@ def simplefilter(action, category=Warning, lineno=0, append=False):
                       "once"), "invalid action: %r" % (action,)
     assert isinstance(lineno, int) and lineno >= 0, \
            "lineno must be an int >= 0"
+    assert isinstance(category, type), "category must be a class"
```

Comment:
I prefer the implementation from the patch by @berkerpeksag:

```
 if not (isinstance(category, type) and issubclass(category, Warning)):
        raise TypeError('category must be a Warning subclass, '
                        'not {!r}'.format(category))
```
You can emulate something close.

You can emulate some parts of the test from the same patch:

```
+    def test_simplefilter_invalid_category(self):
+        class MyWarningClass(Warning):
+            pass
+
+        class NonWarningSubclass:
+            pass
+
+        msg_regex = 'category must be a Warning subclass, not (.*)'
+
+        with self.assertRaisesRegex(TypeError, msg_regex):
+            self.module.simplefilter('always', '')
+
+        with self.assertRaisesRegex(TypeError, msg_regex):
+            self.module.simplefilter('always', NonWarningSubclass)
+
+        with self.assertRaisesRegex(TypeError, msg_regex):
+            self.module.simplefilter('always', MyWarningClass())
```

Reply:
Thank you for the suggestion. I'll take a look at this.

I also have to get some context on the failing tests. Not sure why they are failing.

Reply:
Taking a look at this now. I might update this PR soon.